PostBuild.sh

```diff
@@ -130,51 +130,43 @@ function CreateFstab {
 
 # Configure cloud-init
 function ConfigureCloudInit {
-   local CLOUDCFG
-   local CLINITUSR
-
-   CLOUDCFG="${CHROOTMNT}/etc/cloud/cloud.cfg"
-   CLINITUSR=$( grep -E "name: (maintuser|centos|ec2-user|cloud-user)" \
-            "${CLOUDCFG}" | awk '{print $2}')
-
-   # Reset key parms in standard cloud.cfg file
-   if [ "${CLINITUSR}" = "" ]
-   then
-      err_exit "Astandard cloud-init file: can't reset default-user config"
-   else
-      # Ensure passwords *can* be used with SSH
-      err_exit "Allow password logins to SSH..." NONE
-      sed -i -e '/^ssh_pwauth/s/0$/1/' "${CLOUDCFG}" || \
-        err_exit "Failed allowing password logins"
-
-      # Delete current "system_info:" block
-      err_exit "Nuking standard system_info block..." NONE
-      sed -i '/^system_info/,/^  ssh_svcname/d' "${CLOUDCFG}" || \
-        err_exit "Failed to nuke standard system_info block"
-
-      # Replace deleted "system_info:" block
-      (
-         printf "system_info:\n"
-         printf "  default_user:\n"
-         printf "    name: '%s'\n" "${MAINTUSR}"
-         printf "    lock_passwd: true\n"
-         printf "    gecos: Local Maintenance User\n"
-         printf "    groups: [wheel, adm]\n"
-         printf "    sudo: [ 'ALL=(root) NOPASSWD:ALL' ]\n"
-         printf "    shell: /bin/bash\n"
-         printf "    selinux_user: unconfined_u\n"
-         printf "  distro: rhel\n"
-         printf "  paths:\n"
-         printf "    cloud_dir: /var/lib/cloud\n"
-         printf "    templates_dir: /etc/cloud/templates\n"
-         printf "  ssh_svcname: sshd\n"
-      ) >> "${CLOUDCFG}"
-
-      # Update NS-Switch map-file for SEL-enabled environment
-      err_exit "Enabling SEL lookups by nsswitch..." NONE
-      printf "%-12s %s\n" sudoers: files >> "${CHROOTMNT}/etc/nsswitch.conf" || \
-        err_exit "Failed enabling SEL lookups by nsswitch"
-   fi
+    local CLOUDCFG
+
+    CLOUDCFG="${CHROOTMNT}/etc/cloud/cloud.cfg"
+
+    # Ensure passwords *can* be used with SSH
+    err_exit "Allow password logins to SSH..." NONE
+    sed -i -e '/^ssh_pwauth/s/0$/1/' "${CLOUDCFG}" || \
+    err_exit "Failed allowing password logins"
+
+    # Delete current "system_info:" block
+    err_exit "Nuking standard system_info block..." NONE
+    sed -i '/^system_info/,/^$/d' "${CLOUDCFG}" || \
```

Comment:
Since this now is a "from <TOKEN> to end"  rather than a "from <TOKEN_1> to <TOKEN_2>" range, does this continue to work across all EL8s? Which is to ask, has `/^system_info/` been moved to be the last configuration-block in not ust the OL8 version but all EL8 versions of the file?

I'm going to go look and answer my own question, but figured I'd ask here while I wait for instances to spin up.

Reply:
Ah. Ok. Looking at the one in the official RHEL 8 (you mentioned an "official" Oracle 8, but I'm not finding any such beast), looks like the file ends:

```
system_info:
  default_user:
    name: cloud-user
    lock_passwd: true
    gecos: Cloud User
    groups: [adm, systemd-journal]
    sudo: ["ALL=(ALL) NOPASSWD:ALL"]
    shell: /bin/bash
  distro: rhel
  paths:
    cloud_dir: /var/lib/cloud
    templates_dir: /etc/cloud/templates
  ssh_svcname: sshd

# vim:syntax=yaml
```

My vague recollection is that I'd originally chosen the range-token method because I wanted to preserve the content after the `system_info:` stanza, even though it's, effectively, null.

Reply:
Nor does Alma:
```
$ sed -n '/^system_info/,$p' /etc/cloud/cloud.cfg
system_info:
   # This will affect which distro class gets used
   distro: almalinux
   # Default user name + that default users groups (if added/used)
   default_user:
     name: ec2-user
     lock_passwd: True
     gecos: almalinux Cloud User
     groups: [adm, systemd-journal]
     sudo: ["ALL=(ALL) NOPASSWD:ALL"]
     shell: /bin/bash
   # Other config here will be given to the distro class and/or path classes
   paths:
      cloud_dir: /var/lib/cloud/
      templates_dir: /etc/cloud/templates/
   ssh_svcname: sshd
```

Reply:
Looking at the cloud.cfg in the _base_ AMI:

```
system_info:
  default_user:
    name: ec2-user
    lock_passwd: true
    gecos: Cloud User
    groups: [wheel, adm, systemd-journal]
    sudo: ["ALL=(ALL) NOPASSWD:ALL"]
    shell: /bin/bash
  distro: rhel
  paths:
    cloud_dir: /var/lib/cloud
    templates_dir: /etc/cloud/templates
  ssh_svcname: sshd
```

Meaing the previous `sed`  statement ought to have worked. Updating the instance, now, to see if the cloud.cfg content has changed…

Reply:
I don't think the problem can be reproduced outside of using chroot or similar to create a fresh new image.